src/Twig/MercureExtension.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Twig;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Mercure\Authorization;
+use Symfony\Component\Mercure\HubRegistry;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+/**
+ * Registers the Twig helper function.
+ *
+ * @author Kévin Dunglas <kevin@dunglas.fr>
+ */
+final class MercureExtension extends AbstractExtension
+{
+    private HubRegistry $hubRegistry;
+    private ?Authorization $authorization;
+    private ?RequestStack $requestStack;
+
+    public function __construct(HubRegistry $hubRegistry, ?Authorization $authorization = null, ?RequestStack $requestStack = null)
+    {
+        $this->hubRegistry = $hubRegistry;
+        $this->authorization = $authorization;
+        $this->requestStack = $requestStack;
+    }
+
+    public function getFunctions(): array
+    {
+        return [new TwigFunction('mercure', [$this, 'mercure'])];
+    }
+
+    /**
+     * @param string|string[]|null $topics A topic or an array of topics to subscribe for. If this parameter is omitted or `null` is passed, the URL of the hub will be returned (useful for publishing in JavaScript).
```

Comment:
The url is always returned, isn't it? 
It's probably worth adding an `@return` annotation btw

Reply:
Done